int32 num_x
int32 num_y
int32 num_z
float32 size_x
float32 size_y
float32 size_z
